base/libc.jl

```diff
@@ -170,13 +170,13 @@ library.
 strftime(t) = strftime("%c", t)
 strftime(fmt::AbstractString, t::Real) = strftime(fmt, TmStruct(t))
 function strftime(fmt::AbstractString, tm::TmStruct)
-    timestr = Vector{UInt8}(128)
+    timestr = Base.StringVector(128)
     n = ccall(:strftime, Int, (Ptr{UInt8}, Int, Cstring, Ptr{TmStruct}),
               timestr, length(timestr), fmt, &tm)
     if n == 0
         return ""
     end
-    return String(timestr[1:n])
+    return String(resize!(timestr,n))
```

Comment:
This would make all strings from `strftime` take up 128 bytes instead of what they actually need right? Not sure it is a big deal.

Reply:
That doesn't change the underlying data. A 128 length array resized to length 0 will still occupy the original 128 elements in memeory.

Reply:
So you saved an allocation here, but memory usage went up a factor of ~5, AFAIU.

Reply:
You mean since in the original version the 128 byte buffer can be reclaimed and only the resulting smaller string object is left around, whereas in this version only the former is allocated but therefore it can't be reclaimed. True.

Reply:
Yeah, it probably doesn't matter at all but since one of the reasons for this PR was to reduce an allocation, I thought it made sense to point out that there was also a drawback.

Reply:
Hmm. Probably better to revert this then, but not really a big deal either way.